security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -244,8 +253,9 @@ func (sc *SecretCache) GenerateSecret(ctx context.Context, connectionID, resourc
 				conIDresourceNamePrefix, err)
 			return nil, err
 		}
-		// This is not stored - envoy will refresh when the cert is about to expire.
-		cacheLog.Infoa("GenerateSecret from file", resourceName)
+		// TODO(JimmyCYJ): need a file watcher to detect file updates and push new secret to clients.
```

Comment:
@lei-tang To confirm: we need a file watcher here, right?

Reply:
There is a TODO at line 237 in this file https://github.com/istio/istio/blob/32087603be189c49dcc39f94c46554f0adc4e98c/security/pkg/nodeagent/cache/secretcache.go#L228

Reply:
Thanks.